lib/Menu.php

```diff
@@ -95,10 +94,148 @@ class Menu extends CoreEntity {
+		 */
+		$args = apply_filters( 'wp_nav_menu_args', $args );
+		$args = (object) $args;
+
+		/**
+		 * Since Timber doesn't use HTML here, try to unserialize the maybe cached menu object
+		 *
+		 * @see wp_nav_menu()
+		 */
+		$nav_menu = apply_filters( 'pre_wp_nav_menu', null, $args );
+		if( null !== $nav_menu ) {
+			try {
+				$nav_menu = unserialize($nav_menu);
+				if( $nav_menu instanceof Menu ) {
+					return $nav_menu;
+				}
+			} catch(\Throwable $e) {
+			}
+		}
+
+		// No menu term provided, get the pages menu
+		if(!$menu) {
+
+			$defaults = [
+				'sort_column'  => 'menu_order, post_title',
+				'menu_id'      => '',
+				'menu_class'   => 'menu',
+				'container'    => 'div',
+				'echo'         => true,
+				'link_before'  => '',
+				'link_after'   => '',
+				'before'       => '<ul>',
+				'after'        => '</ul>',
+				'item_spacing' => 'discard',
+				'walker'       => '',
+			];
+
+			$args = wp_parse_args( (array) $args, $defaults );
+
+			/**
+			 * @see wp_page_menu()
+			 */
+			$args = apply_filters( 'wp_page_menu_args', $args );
+
+			$nav_menu = new static(null, $args);
+			$nav_menu->init_pages_menu();
+
+			/**
+			 * @see wp_page_menu()
+			 */
+			$_nav_menu = apply_filters( 'wp_page_menu', serialize($nav_menu), $args );
+
+			return $nav_menu;
+		}
+
+		// Skip the menu term guessing part, we already have our menu term
+
+		$menu_items = wp_get_nav_menu_items( $menu->term_id, ['update_post_term_cache' => false]);
+
+		_wp_menu_item_classes_by_context($menu_items);
+
+		$sorted_menu_items        = [];
+		$menu_items_with_children = [];
+		foreach ( (array) $menu_items as $menu_item ) {
+			$sorted_menu_items[ $menu_item->menu_order ] = $menu_item;
+			if ( $menu_item->menu_item_parent ) {
+				$menu_items_with_children[ $menu_item->menu_item_parent ] = true;
+			}
+		}
+
+		// Add the menu-item-has-children class where applicable.
+		if ( $menu_items_with_children ) {
+			foreach ( $sorted_menu_items as &$menu_item ) {
+				if ( isset( $menu_items_with_children[ $menu_item->ID ] ) ) {
+					$menu_item->classes[] = 'menu-item-has-children';
+				}
+			}
+		}
+
+		unset( $menu_items, $menu_item );
+
+		/**
+		 * @see wp_nav_menu()
+		 */
+		$sorted_menu_items = apply_filters( 'wp_nav_menu_objects', $sorted_menu_items, $args );
+
+		// Create Menu object
+		$nav_menu = new static($menu, (array) $args);
+		$nav_menu->sorted_menu_items = $sorted_menu_items;
+
+		// Convert items into MenuItem objects
+		$sorted_menu_items = $nav_menu->convert_menu_items($sorted_menu_items);
+		$sorted_menu_items = $nav_menu->order_children($sorted_menu_items);
+		$sorted_menu_items = $nav_menu->strip_to_depth_limit($sorted_menu_items);
+
+		$nav_menu->items = $sorted_menu_items;
+		unset( $sorted_menu_items );
+
+		/**
+		 * Since Timber doesn't use HTML, serialize the menu object to provide a cacheable string
+		 *
+		 * @see wp_nav_menu()
+		 */
+		$_nav_menu = apply_filters( 'wp_nav_menu', serialize($nav_menu), $args );
```

Comment:
Caching menu plugins are [misusing this filter](https://github.com/inpsyde/menu-cache/blob/master/index.php#L43-L44) to cache menus and excepts a string here, provide a serialized menu object.